server/src/tree-sitter/analyzer.ts

```diff
@@ -707,62 +733,133 @@ export default class Analyzer {
     }
   }
 
-  public processRecipeScanResults (scanResults: string, uri: any): RequestResult['ProcessRecipeScanResults'] {
-    let uriString
+  /**
+   *
+   * @param scanResult
+   * @param uri
+   *
+   * Process scan results sent from the client. The scan results are generated by bitbake -e <recipe> command
+   */
+  public processRecipeScanResults (scanResult: string, uri: any): RequestResult['ProcessRecipeScanResults'] {
```

Comment:
I suggest refactoring this function into smaller splits, it has too many responsibilities.

Reply:
Will do it after others are resolved.